contracts/MandatoryLockupDistributor.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import "OpenZeppelin/openzeppelin-contracts@4.6.0/contracts/token/ERC20/IERC20.sol";
+import "OpenZeppelin/openzeppelin-contracts@4.6.0/contracts/utils/cryptography/MerkleProof.sol";
+
+interface IOGVStaking {
+    function stake(
+        uint256 amount,
+        uint256 end,
+        address to
+    ) external;
+}
+
+contract MandatoryLockupDistributor {
+    //@notice This event is triggered whenever a call to #claim succeeds.
+    event Claimed(uint256 index, address account, uint256 amount);
+
+    address public immutable token;
+    bytes32 public immutable merkleRoot;
+    address public immutable stakingContract;
+
+    // This is a packed array of booleans.
+    mapping(uint256 => uint256) private claimedBitMap;
+
+    constructor(
+        address _token,
+        bytes32 _merkleRoot,
+        address _stakingContract
+    ) {
+        token = _token;
+        merkleRoot = _merkleRoot;
+        stakingContract = _stakingContract;
+    }
+
+    /**
+     * @dev
+     * @param _index Index in the tree
+     */
+    function isClaimed(uint256 _index) public view returns (bool) {
```

Comment:
@tomlinton What's the purpose of this index bitmasking dance vs just reading and writing a boolean to a mapping?

Reply:
It's a fork of the uniswap merkle distributor which seems easier than rewriting it. I'd guess it's also more gas efficient.